packages/block-library/src/navigation-menu/index.php

```diff
@@ -15,20 +15,69 @@
  * @return string Returns the post content with the legacy widget added.
  */
 function render_block_navigation_menu( $attributes, $content, $block ) {
-	return '<nav class="wp-block-navigation-menu">' . build_navigation_menu_html( $block ) . '</nav>';
+	// Add CSS classes and inline styles.
+	$bg_color_css_classes = '';
+	if ( array_key_exists('backgroundColor', $attributes ) ) {
+		$bg_color_css_classes .= ' has-background-color';
+	}
+	if ( array_key_exists('backgroundColorCSSClass', $attributes ) ) {
+		$bg_color_css_classes .= " {$attributes['backgroundColorCSSClass']}";
+	}
+	$bg_color_css_classes = trim( $bg_color_css_classes );
+
+	$text_color_css_classes = '';
+	if ( array_key_exists('textColor', $attributes ) ) {
+		$text_color_css_classes .= ' has-text-color';
+	}
+
+	if ( array_key_exists('textColorCSSClass', $attributes ) ) {
+		$text_color_css_classes .= " ${attributes['textColorCSSClass']}";
+	}
+	$text_color_css_classes = trim( $text_color_css_classes );
+
+
+	$bg_inline_styles = '';
+	if ( array_key_exists('customBackgroundColor', $attributes ) ) {
+		$bg_inline_styles = "background-color: ${attributes['customBackgroundColor']};";
+	} elseif ( array_key_exists('backgroundColorValue', $attributes ) ) {
+		$bg_inline_styles = "background-color: ${attributes['backgroundColorValue']};";
```

Comment:
Can you help me understand the difference between `customBackgroundColor` and `backgroundColorValue`?

Reply:
I suspect `backgroundColorValue` is one of the defined colors (picked up from the Theme) which display as swatches in the UI. 

`customBackgroundColor` probably comes into play when the user wants to pick a totally custom color using a color picker.

Am I right @retrofox ?

Reply:
Fair question.

> Am I right @retrofox ?
Partially 👍, but I'm the resonsable of the confusion here.

> I suspect backgroundColorValue is one of the defined colors (picked up from the Theme) which display as swatches in the UI.

Actually it's the `backgroundColor` one:

<img src="https://user-images.githubusercontent.com/77539/66920146-1f262700-eff9-11e9-8f24-ee12b0578fcb.png" width="250" />

> customBackgroundColor probably comes into play when the user wants to pick a totally custom color...

![image](https://user-images.githubusercontent.com/77539/66920280-5563a680-eff9-11e9-8045-6f29dc0f0a48.png)

👏 


`backgroundColor` and `customBackgroundColor` (as well as `textColor` and `customTextColor`) are defined and handled by the `withColors()` HOC.
The thing with `backgroundColor` is that its value is defined literaly when the color isn't pick up from the custom ones, for instance `primary`, `has-secondary-background-color`, `has-light-gray-background-color`, etc...
For this reason, I've added the colorValue attibute which **always** will store the color in the RGB notation, useful to define the colors in the inline mode.